pkg/curatedpackages/kubeclient.go

```diff
@@ -0,0 +1,57 @@
+package curatedpackages
+
+import (
+	"os"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	packagesv1 "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+)
+
+// NewKubeClientFromFilename creates a controller-runtime k8s client for use
+// by CLI commands.
+//
+// If the RESTConfigurator is nil, a default configurator will be used.
+func NewKubeClientFromFilename(kubeConfigFilename string, rc RESTConfigurator) (client.Client, error) {
```

Comment:
I would love to avoid passing nil, we are not really representing the abscence of data but the intent to not alter the default config. Also, I think it makes the caller code more difficult to read. 

We could use a builder pattern (maybe with closures as options). Or alternatively have a `NewKubeClientFromFilenameWithDefaultConfigurator(kubeConfigFilename string)` that just calls `NewKubeClientFromFilename`.

Reply:
Yeah, I guess I'm just used to passing nil in constructors for things you don't need/want. You've brought this up before, so clearly you're not. :)

I'd argue that, especially for internal-facing code, where the only intended consumers are ourselves, this is fine. Clear documentation makes up for it. It's certainly simpler and easier to grok than closure-based initializers, with their multiple hops of indirection and abstraction.

So especially given that we've only got a single thing to pass or not right now, I'll elect to go with NewKubeClientFromFilenameWithOptions(kubeConfigFilename string, rc RESTConfigurator) and NewKubeClientFromFilename(kubeConfigFilename string) as the two choices. Hopefully that strikes a nice compromise between our views on the matter?

Reply:
Having now made these changes, I wonder if we're really doing the consumer any favors. They now have two different constructors to choose between. We haven't really made anything simpler for them.

All the "simpler" of the two constructors does is immediately call the other constructor, which I guess is a strong implication that the RESTConfigurator argument can be nil, which ... I guess is good, but I don't know that it's better than the simplicity of having one constructor with a decent comment explaining its usage. I don't feel like having to read a godoc to use some facility is a terrible thing.

All that said, I'm fine with either approach if you are.

Reply:
I prefer the two methods than passing a nil argument :)
But I do understand your concern about having two constructors to choose from

A third alternative would be to have a public setter to update the configurator after initialization. That would mean we would need to separate construction from initialization, like `NewClient` + `client.Init`

Reply:
We had a bit of a discussion in Slack and settled on a private constructor that allows for the restConfigurator to be specified, but a single exported constructor that doesn't. This satisfies both of our desires for a simple clean interface, while still giving us testable code. Win-win I think!